src/server/mongodb/actions/Test.js

```diff
@@ -1,4 +1,4 @@
-import connectMongo from "../../utils/connectMongo.js"
+import connectMongo from "../connectMongo"
```

Comment:
Ideally the functions in this file wouldn't need to repeatedly call `connectMongo` (just once in the top-level), since the connection should be established once and cached.

Reply:
The connection is indeed be established once and cached to use for any other modules. But the connection must be established at least once so it can be cached, at this stage I cannot ensure the connection is established, hence I store cached db connection within global variable and connectMongo will return back cached connection if it exist instead of establish new connection, thelriodicaly you can access cached  connection by calling global variable `global.mongoose` If you are certain that connection have indeed been established at least once.

Reply:
My implementation for calling `connectMongo` is to retrieve cached connection, not try to connect it again

Reply:
Top level await will trigger a unknown issue with web-pack runtime error `[TypeError: Cannot read properties of undefined (reading 'split')](https://stackoverflow.com/questions/73537481/how-can-i-fix-typeerror-cannot-read-properties-of-undefined-reading-split-i)`